popgen/mcmcSelection.go

```diff
@@ -122,13 +123,28 @@ func MetropolisHastings(data AFS, muZero float64, sigmaZero float64, iterations
 	}
 
 	maxN := findMaxN(data)
-	binomMap := make([][]float64, maxN+1)
+	allN := findAllN(data)
+	alleleFrequencyCache = cacheIndexToAlleleFrequencyCache(bins)
+	nkpCache := make([][][]float64, maxN+1)
+	var coefficient float64
+
+	for n := 0; n < len(allN); n++ {
+		nkpCache[allN[n]] = make([][]float64, allN[n])
+		for k := 0; k < len(allN[n]); k++ {
```

Comment:
Should "k" start at one?

Reply:
the argument here is we don't need to initialize these values because all variants have a frequency of at least k = 1? Am I understanding correctly?